.github/workflows/ci_macos.yml

```
@@ -0,0 +1,137 @@
+          brew install coreutils
+          sudo ln -s /usr/local/bin/gtimeout /usr/local/bin/timeout
+          sudo ln -s /opt/homebrew/bin/greadlink /usr/local/bin/greadlink
+          which timeout
+          which greadlink
+          sudo mkdir /usr/local/include
+          ls /usr/local
+          
+      - name: Install Fuse-t and dependencies
+        run: |
+          brew install python-setuptools
+          wget -q https://github.com/macos-fuse-t/fuse-t/releases/download/1.0.44/fuse-t-macos-installer-1.0.44.pkg
+          sudo installer -pkg fuse-t-macos-installer-1.0.44.pkg -target /
+          ls /usr/local/include
+
+      - name: Build CVMFS
+        run: |
+          cd ${GITHUB_WORKSPACE}
+          mkdir -p build
+          ./ci/build_package.sh ${GITHUB_WORKSPACE} ${GITHUB_WORKSPACE}/build cvmfs
+
+      - name: Setup firmlinks
+        run: |
+          # / is readonly on macos 11+ - do 'synthetic firmlink' to create /cvmfs
+          sudo zsh -c 'echo -e "cvmfs\tUsers/Shared/cvmfs\n#comment\n" > /etc/synthetic.conf'
+          sudo chown root:wheel /etc/synthetic.conf
+          sudo chmod a+r /etc/synthetic.conf
+          # apfs.util seems to return non-zero error codes also on success
+          sudo /System/Library/Filesystems/apfs.fs/Contents/Resources/apfs.util -t || true
+
+          
+
+      - name: Install CVMFS
+        run: |
+          export CVMFS_PACKAGE=$(ls ${GITHUB_WORKSPACE}/build/pkg_result/cvmfs-*.pkg)
+          sudo installer -pkg ${CVMFS_PACKAGE} -target /
+
+      - name: Setup CVMFS
+        run: |
+          # / is readonly on macos 11+ - do 'synthetic firmlink' to create /cvmfs
+          sudo zsh -c 'echo -e "cvmfs\tUsers/Shared/cvmfs\n#comment\n" > /etc/synthetic.conf'
+          sudo chown root:wheel /etc/synthetic.conf
+          sudo chmod a+r /etc/synthetic.conf
+          # apfs.util seems to return non-zero error codes also on success
+          sudo /System/Library/Filesystems/apfs.fs/Contents/Resources/apfs.util -t || true
+          sudo cvmfs_config setup
+          sudo cvmfs_config chksetup || exit 1
+
+      - name: Run Tests
+        run: |
+          sudo launchctl limit maxfiles 65536 65536 
+          export CVMFS_PLATFORM_NAME="macos"
+          export CVMFS_TIMESTAMP=$(date -u +'%Y-%m-%dT%H:%M:%SZ')
+          export CVMFS_TEST_CLASS_NAME=ClientIntegrationTests
+          export CVMFS_TEST_PROXY=DIRECT
+          cd $GITHUB_WORKSPACE/test
+          ./run.sh /tmp/cvmfs-client-test.log -o /tmp/cvmfs-client-test.xunit.xml \
+            -s quick                                          \
+            -x src/004-davinci                                \
+            src/005-asetup                                    \
+            src/006-buildkernel                               \
+            src/007-testjobs                                  \
+            src/008-default_domain                            \
+            src/016-dnsunreachable                            \
```

```suggestion
            src/015-rebuild_on_crash                           \
            src/016-dnsunreachable                            \
```
Let's move test 15 to an issue